### PR TITLE
fix: Abstract WORDSIZE to specify maxwidth on qasm conversion

### PR DESCRIPTION
**Pull request by qartik**
# Description

Makes pytket-phir compatible with pytket version 1.30.0 while maintaining backward compatibility.

Fixes: #203

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# Checklist

- [x] I have performed a self-review of my code
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog with any user-facing changes

### CODE REVIEW

File: pytket/phir/api.py [hunk 1]
```diff
@@ -100,10 +100,12 @@ def qasm_to_phir(
             wasm_file.flush()
             wasm_file.close()
 
-            circuit = circuit_from_qasm_wasm(qasm_file.name, wasm_file.name)
+            circuit = circuit_from_qasm_wasm(
+                qasm_file.name, wasm_file.name, maxwidth=WORDSIZE
```
Comment (qartik) [inline pytket/phir/api.py:104]: @dlucchetti do you know if there any implications to using 64 bits as `maxwidth` here since WASM only supports 32 bits?

**Review by cqc-alec — DISMISSED** [review]

File: pytket/phir/api.py (hunk 1 above)
Comment (peter-campora) [reply pytket/phir/api.py:104]: I haven't tried this, but has anyone seen what happens when you compile a function taking a 64-bit parameter from Rust to wasm, create a >32 bit creg and write into the upper bits beyond 2^32 before passing the creg to the wasm function?

Comment (dlucchetti) [reply pytket/phir/api.py:104]: Without special handling, I would expect this to diverge from hardware behavior.  Perhaps easiest way could be disallowing passing of registers that are greater than 32 bits?

Comment (qartik) [reply pytket/phir/api.py:104]: Makes sense, I currently see an error thrown later in the stack if I pass registers of size 33:
```py
tests/test_wasm.py:85:
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = [], funcname = 'multi'
filehandler = Functions in wasm file with the uid 3138ec2df84e13dcee5f3772555e93d43de8f2e2a0937770c5959bca2da4fb10:
function 'init' ...32 parameter(s) and 1 i32 return value(s)
unsupported function with invalid parameter or result type: 'add_something'

list_i = [BitRegister("c0", 33), BitRegister("c1", 43)], list_o = [BitRegister("c2", 53)], args_wasm = [0], kwargs = {}, reg = BitRegister("c0", 33)

    def overload_add_wasm_to_reg(
        self: Circuit,
        funcname: str,
        filehandler: wasm.WasmFileHandler,
        list_i: Sequence[BitRegister],
        list_o: Sequence[BitRegister],
        args_wasm: Optional[Sequence[int]] = None,
        **kwargs: Any,
    ) -> Circuit:
        """Add a classical function call from a wasm file to the circuit.
        \n\n:param funcname: name of the function that is called
        \n:param filehandler: wasm file handler to identify the wasm file
        \n:param list_i: list of the classical registers assigned to
         the input variables of the function call
        \n:param list_o: list of the classical registers assigned to
         the output variables of the function call
        \n:param args_wasm: vector of wasmstates the wasm op should be added to
        \n:param kwargs: additional arguments passed to `add_gate_method` .
         Allowed parameters are `opgroup`,  `condition` , `condition_bits`,
         `condition_value`
        \n:return: the new :py:class:`Circuit`"""

        if args_wasm is None:
            args_wasm = [0]

        if filehandler._check_file:
            for reg in list_i:
                if reg.size > 32:
>                   raise ValueError(
                        """wasm is only supporting 32 bit size registers,
    please use only registers of at most 32 bits"""
                    )
E                   ValueError: wasm is only supporting 32 bit size registers,
E                   please use only registers of at most 32 bits

.venv/lib/python3.12/site-packages/pytket/circuit/__init__.py:141: ValueError
```

Perhaps we can raise a similar error early from pytket-phir.

**Review by dlucchetti — APPROVED** [review]

File: pyproject.toml [hunk 2]
```diff
@@ -31,7 +31,7 @@ dependencies = [
 
 [project.optional-dependencies]
 docs = ["sphinx", "pydata_sphinx_theme"]
-phirc = ["projectq", "quantum-pecos>=0.5.0.dev10"]
+phirc = ["projectq", "quantum-pecos>=0.6.0.dev2"]
```
Comment (qartik) [inline pyproject.toml:34]: 64-bit support added in https://github.com/PECOS-packages/PECOS/releases/tag/v0.6.0.dev2

File: pytket/phir/phirgen.py [hunk 3]
```diff
@@ -544,7 +556,7 @@ def get_decls(qbits: set["Qubit"], cbits: set[tkBit]) -> list[dict[str, str | in
     decls += [
         {
             "data": "cvar_define",
-            "data_type": "u32",
+            "data_type": f"i{WORDSIZE}",
```
Comment (qartik) [inline pytket/phir/phirgen.py:559]: Changing to i64 as the default as per the spec. Tested with previous version of pytket to check that i32 works as well.

**Review by cqc-alec — APPROVED** [review]

**Review by dlucchetti — APPROVED** [review]

**Pull request merged**